### ISSUE TITLE
Non-working install when installing from AUR packages

### ISSUE THREAD

**Issue by Smile4ever** [KELiON/cerebro#127]
On Arch Linux, when installing the cerebro and cerebro-git AUR packages result in non-working install. I get the cerebro window, but I can not type in it:
![cerebro-arch](https://cloud.githubusercontent.com/assets/544411/23182342/84447978-f878-11e6-948a-b6033484c228.png)

If I run cerebro in a terminal, I get no error whatsoever.

AppImage is working fine though.

**Comment by maximbaz** [comment]
I confirm, looks like something got changed in the tools that we use to build cerebro. Both packages worked fine when `0.2.6` was released, but now there is a build failure *if you clone a repo from scratch*. And that is what both AUR packages do, they download sources and build it from scratch.

@KELiON you might want to look into this as well. I have a cloned `cerebro` repo since a few months, I do a `git pull` and then try to build - everything is fine. But when I clone the repo from scratch and try to build it, then I get a build failure.

Repro steps:

```
git clone https://github.com/KELiON/cerebro.git cerebro2
cd cerebro2
npm install
cd app/ 
npm install
cd ../

npm run build
```

The last command worked earlier and still works on my old-cloned `cerebro` repository, but it fails on the `cerebro2` that I've just cloned above with the following errors:

```
ERROR in ./app/main/plugins/core/maps/index.js
Module parse failed: /tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/babel-loader/index.js!/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/app/main/plugins/core/maps/index.js Unexpected token (34:63)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (34:63)
    at Parser.pp$4.raise (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:2221:15)
    at Parser.pp.unexpected (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:603:10)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1822:12)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1608:25)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1861:32)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1861:32)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseFunctionBody (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:2098:24)
    at Parser.pp$3.parseArrowExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:2087:10)
    at Parser.pp$3.parseParenArrowList (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1902:17)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1870:21)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
 @ ./app/main/plugins/core/index.js 12:8-25



ERROR in ./app/main/plugins/core/files/getPreview/preview/Archive.js
Module parse failed: /tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/babel-loader/index.js!/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/app/main/plugins/core/files/getPreview/preview/Archive.js Unexpected token (41:48)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (41:48)
    at Parser.pp$4.raise (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:2221:15)
    at Parser.pp.unexpected (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:603:10)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1822:12)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1608:25)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1861:32)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1608:25)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1861:32)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
    at Parser.pp$3.parseParenAndDistinguishExpression (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1861:32)
    at Parser.pp$3.parseExprAtom (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1796:19)
    at Parser.pp$3.parseExprSubscripts (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1692:19)
    at Parser.pp$3.parseExprOps (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1637:21)
    at Parser.pp$3.parseMaybeConditional (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1620:21)
    at Parser.pp$3.parseMaybeAssign (/tmp/yaourt-tmp-maximbaz/aur-cerebro/src/cerebro-0.2.6/node_modules/acorn/dist/acorn.js:1597:21)
 @ ./app/main/plugins/core/files/getPreview/index.js 59:15-43

```

**Comment by maximbaz** [comment]
Experiment has shown that the issue is caused by the latest release `6.23.0` of babel. Even though dependency versions are locked, the _nested dependency versions_ are not.

On the working repo:

```
├─┬ babel-core@6.11.4
│ ├─┬ babel-code-frame@6.22.0
│ │ └── js-tokens@3.0.1
│ ├─┬ babel-generator@6.22.0
│ │ ├─┬ detect-indent@4.0.0
│ │ │ └─┬ repeating@2.0.1
│ │ │   └── is-finite@1.0.2
│ │ └── jsesc@1.3.0
│ ├── babel-helpers@6.22.0
│ ├── babel-messages@6.22.0
│ ├─┬ babel-runtime@6.22.0
```

On the newly cloned repo:
```
├─┬ babel-core@6.11.4
│ ├─┬ babel-code-frame@6.22.0
│ │ └── js-tokens@3.0.1
│ ├─┬ babel-generator@6.23.0
│ │ ├─┬ detect-indent@4.0.0
│ │ │ └─┬ repeating@2.0.1
│ │ │   └── is-finite@1.0.2
│ │ ├── jsesc@1.3.0
│ │ └── trim-right@1.0.1
│ ├── babel-helpers@6.23.0
│ ├── babel-messages@6.23.0
│ ├─┬ babel-runtime@6.23.0
```

@KELiON could you perhaps use [npm shrinkwrap](https://docs.npmjs.com/cli/shrinkwrap) to lock the versions of the all the nested dependencies? That way we will not hit any surprises like that in the future. Ideally do it on the repo which doesn't have babel `6.23.0` yet :)

**Comment by KELiON** [comment]
@maximbaz I'll check why it fails on babel 6.23.0 and try to fix it. And I'd just migrate to yarn, that should solve all npm problems. 

I thought that fixing versions in `package.json` is strict enough to avoid this kind of problems, but it is not!  Thanks for investigation, @maximbaz

**Comment by KELiON** [comment]
@maximbaz I hope #148 should fix all npm-related shit. Can you try to install Cerebro from scratch from this branch on your machine and let me know if it is ok?

**Comment by maximbaz** [comment]
Just tested, works beautifully! Thanks a lot! 

For people interested in AUR packages, I can update the build commands for `cerebro-git` right after #148 is merged into master, but to get the regular `cerebro` we'd need to wait for the next release.

**Comment by jorgicio** [comment]
The Yarn update works flawlessly. I maintain the package for Gentoo in my overlay, used yarn instead of npm, and it works fine.

**Comment by KELiON** [comment]
@maximbaz #148 is merged, you can update `cerebro-git`

**Comment by maximbaz** [comment]
Thanks! `cerebro-git` is working again.

**Comment by maximbaz** [comment]
The `cerebro` on AUR is updated to 0.2.7, you can get back to stable package if you want :tada: 

/cc @Smile4ever, @Godod, @pschmitt, @Cross03, @quatroka, @hughwilliams94, @commanderkelso, @sava-vidakovic, @harryio, @tunnckoCore, @mijoharas, @m1aw.

**Comment by tunnckoCore** [comment]
@maximbaz great! :tada:
